### ISSUE TITLE
About the weight normalization?

### ISSUE THREAD

**Issue by PkuRainBow** [pytorch/pytorch#5865]
I have one question about the implementation for 

`nn.utils.weight_norm()`

I read the doc, it says that, 

>   Weight normalization is a reparameterization that decouples the magnitude
>     of a weight tensor from its direction. This replaces the parameter specified
>     by `name` (e.g. "weight") with two parameters: one specifying the magnitude
>     (e.g. "weight_g") and one specifying the direction (e.g. "weight_v").

In fact, I want to normalize the weight vectors(L2 norm=1), so I am wondering whether the following function call can ensure that the weights' L2 norm equal 1 or it will learn a magnitude factor "weight_g"? So How can I ensure that the "weight_g" equal 1?

`nn.utils.weight_norm(nn.Conv2d(512, num_classes, kernel_size=1, stride=1, padding=1, dilation=1, bias=False), name='weight')`

**Comment by SsnL** [comment]
It parametrizes w as w=g * v/||v|| without posing any constraint on either g, v, or w. 

If you want to ensure that w has magnitude 1, I suggest doing one of the two approaches:
1. reparameterize as w = g / ||g||
2. do a manual projection after every optimization step.
